src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'green' | 'red' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
 
-}
+  constructor(
+    public color: 'green' | 'red' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error();
+    }
+
+    const biggestSide = Math.max(this.a, this.b, this.c);
+
+    if (biggestSide >= this.b + this.c + this.a - biggestSide) {
+      throw new Error();
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter = (this.a + this.b + this.c) / 2;
 
-export class Circle {
+    const area
+      = Math.floor(((semiperimeter * (semiperimeter - this.a)
+      * (semiperimeter - this.b) * (semiperimeter - this.c))
+      ** 0.5) * 100) / 100;
 
+    return area;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
 
+  constructor(
+    public color: 'green' | 'red' | 'blue',
+    public radius: number,
+  ) {
+    this.shape = 'circle';
+
+    if (this.radius <= 0) {
+      throw new Error();
```

Comment:
Error message is not descriptive. It should provide more information about the error.